libs/hdf-converters/src/converters-from-hdf/caat/reverse-caat-mapper.ts

```diff
@@ -0,0 +1,257 @@
+  filename?: string;
+  controls?: ContextualizedControl[];
+};
+
+type Data = InputData & {data: ContextualizedEvaluation; filename: string};
+
+export const CAATHeaders = [
+  'Control Number',
+  'Finding Title',
+  'Date Identified',
+  'Finding ID',
+  'Information System or Program Name',
+  'Repeat Findings',
+  'Repeat Finding Weakness ID',
+  'Finding Description',
+  'Weakness Description',
+  'Control Weakness Type',
+  'Source',
+  'Assessment/Audit Company',
+  'Test Method',
+  'Test Objective',
+  'Test Result Description',
+  'Test Result',
+  'Recommended Corrective Action(s)',
+  'Effect on Business',
+  'Likelihood',
+  'Impact'
+] as const;
+
+type CAATRow = Partial<
+  Record<(typeof CAATHeaders)[number], string | undefined>
+>;
+
+export class FromHDFToCAATMapper {
+  static readonly MaxCellSize = 32000;
+
+  static readonly NistCanonizationConfig: CanonizationConfig = {
+    max_specifiers: 3,
+    pad_zeros: true,
+    allow_letters: false,
+    add_spaces: false
+  };
+
+  static readonly FileSettings: XLSX.Properties = {
+    Title: 'Compliance Assessment/Audit Tracking (CAAT) Spreadsheet',
+    Subject: 'Assessment Data',
+    Author: 'MITRE SAF',
+    CreatedDate: new Date()
+  };
+
+  static readonly SheetOptions: XLSX.JSON2SheetOpts = {
+    header: CAATHeaders.slice() // CAATHeaders is immutable but the type expects a mutable string
+  };
+
+  static formatDate(date: Date, delimiter: string): string {
+    return [
+      Intl.DateTimeFormat('en-US', {month: '2-digit'}),
+      Intl.DateTimeFormat('en-US', {day: '2-digit'}),
+      Intl.DateTimeFormat('en-US', {year: 'numeric'})
+    ]
+      .map((formatter) => formatter.format(date))
+      .join(delimiter);
+  }
+
+  data: Data[];
+
+  // ensure input is turned into an array of contextualized evaluations with some additional metadata
+  constructor(data: InputData | InputData[]) {
+    if (!Array.isArray(data)) {
+      data = [data];
+    }
+    this.data = data.map((datum) => {
+      let contextualizedEvaluation = datum.data;
+      if (_.isString(contextualizedEvaluation)) {
+        const contextualizedFile = convertFileContextual(
+          contextualizedEvaluation
+        );
+        if (!isContextualizedEvaluation(contextualizedFile)) {
+          throw new Error('Input string was not an HDF ExecJSON');
+        }
+        contextualizedEvaluation = contextualizedFile;
+      }
+      contextualizedEvaluation = ensureContextualizedEvaluation(
+        contextualizedEvaluation
+      );
+      // if provided a filename use that, otherwise the name of the first profile in the execjson (usually there's either only one or if there are multiple then the one at the top is the wrapper or overlay profile) otherwise if there are no profiles at all then a default of 'ExecJSON'
+      return {
+        data: contextualizedEvaluation,
+        filename:
+          datum.filename ??
+          contextualizedEvaluation.data.profiles.at(0)?.name ??
```

Comment:
```suggestion
          contextualizedEvaluation.data.profiles.at(0)?.name ??      // Otherwise, use the name of the first profile. This is typically the only profile or the wrapper or overlay profile if there are multiple.
```